src/opera/pge/dswx_s1/dswx_s1_pge.py

```diff
@@ -360,6 +363,47 @@ def _geotiff_filename(self, inter_filename):
 
         return f"{tile_filename}_{band_idx}_{band_name}.tif"
 
+    def _browse_filename(self, inter_filename):
+        """
+        Returns the file name to use for the PNG browse image produced by
+        the DSWx-S1 PGE.
+
+        The browse image filename for the DSWx-S1 PGE consists of:
+
+            <Tile filename>_BROWSE.png
+
+        Where <Tile filename> is returned by DSWxS1PostProcessorMixin._tile_filename().
+
+        Parameters
+        ----------
+        inter_filename : str
+            The intermediate filename of the output browse image to generate a
+            filename for. This parameter may be used to inspect the file in order
+            to derive any necessary components of the returned filename.
+
+        Returns
+        -------
+        browse_image_filename : str
+            The file name to assign to browse image created by this PGE.
+
+        """
+        # Find one of the tif files that correspond to the provided browse image
+        # based on tile ID, so we can extract its metadata to derive the tile filename
+        tile_id = basename(inter_filename).split('_')[3]
+
+        tif_pattern = f"*{tile_id}*.tif*"
+
+        tif_files = list(glob.glob(join(dirname(inter_filename), tif_pattern)))
+
+        if not len(tif_files):
+            error_msg = f"Could not find any .tif files corresponding to browse image {inter_filename}"
+            self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+
+        # Use the first found tif file, they should all have the same metadata
```

Comment:
This is another personal thing, but rather than 'first found'  maybe say 'first tif file in the list'.